File: backend/routes/conversations.py
from flask import Blueprint, request, jsonify
from db import chat as chat_db

conv_bp = Blueprint("conversations", __name__)


@conv_bp.route("/api/conversations", methods=["GET"])
def list_conversations():
    convs = chat_db.list_conversations()
    return jsonify(convs)


@conv_bp.route("/api/conversations", methods=["POST"])
def create_conversation():
    body = request.get_json(force=True) or {}
    title = body.get("title")
    conv = chat_db.create_conversation(title)
    return jsonify(conv), 201


@conv_bp.route("/api/conversations/<conv_id>", methods=["GET"])
def get_conversation(conv_id: str):
    conv = chat_db.get_conversation(conv_id)
    if not conv:
        return {"error": "Not found."}, 404
    return jsonify(conv)


@conv_bp.route("/api/conversations/<conv_id>/messages", methods=["GET"])
def get_messages(conv_id: str):
    conv = chat_db.get_conversation(conv_id)
    if not conv:
        return {"error": "Not found."}, 404
    messages = chat_db.get_messages(conv_id)
    return jsonify(messages)

File: frontend/src/api/conversations.ts
import type { Conversation, Message } from './types';

const BASE = '/api';

export async function listConversations(): Promise<Conversation[]> {
  const res = await fetch(`${BASE}/conversations`);
  if (!res.ok) throw new Error('Failed to list conversations');
  return res.json();
}

export async function createConversation(title?: string): Promise<Conversation> {
  const res = await fetch(`${BASE}/conversations`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ title: title ?? null }),
  });
  if (!res.ok) throw new Error('Failed to create conversation');
  return res.json();
}

export async function getConversation(id: string): Promise<Conversation> {
  const res = await fetch(`${BASE}/conversations/${id}`);
  if (!res.ok) throw new Error('Conversation not found');
  return res.json();
}

export async function getMessages(convId: string): Promise<Message[]> {
  const res = await fetch(`${BASE}/conversations/${convId}/messages`);
  if (!res.ok) throw new Error('Failed to fetch messages');
  return res.json();
}
